Tasks/JavaToolInstaller/AzureStorageArtifacts/AzureStorageArtifactDownloader.ts

```diff
@@ -0,0 +1,89 @@
+import * as Q from 'q';
+import * as  tl from 'vsts-task-lib/task';
+import msRestAzure = require("azure-arm-rest/azure-arm-common");
+import Model = require("azure-arm-rest/azureModels");
+import armStorage = require('azure-arm-rest/azure-arm-storage');
+import BlobService = require('azure-blobstorage-artifactProvider/blobservice');
+
+export class AzureStorageArtifactDownloader {
+  public connectedService: string;
+  public azureStorageAccountName: string;
+  public containerName: string;
+  public commonVirtualPath: string;
+
+
+  constructor() {
+    this.connectedService = tl.getInput('ConnectedServiceNameARM', true);
+    this.azureStorageAccountName = tl.getInput('storageAccountName', true);
+    this.containerName = tl.getInput('containerName', true);
+    this.commonVirtualPath = tl.getInput('commonVirtualPath', false);
+  }
+
+  public async downloadArtifacts(downloadToPath: string, fileEnding: string): Promise<void> {
+    console.log(tl.loc('DownloadFromAzureBlobStorage', this.containerName));
+
+    let storageAccount: StorageAccountInfo = await this._getStorageAccountDetails();
```

Comment:
Use `const` instead of `let` when possible.  I think most `let`s in the PR could be `const`

Reply:
Good point. I'll update them.